src/network/connection.cpp

```diff
@@ -710,6 +728,32 @@ void Connection::update() {
 			}
 
 			break;
+		}
+
+		case PACKET_SET_CONFIG_FLAG: {
+			// Packet type (4) + Packet number (8) + sensor_id(1) + flag_id (2) + state (1)
+			if (len < 16) {
+				m_Logger.warn("Invalid sensor config flag packet: too short");
+				break;
+			}
+			uint8_t sensorId = m_Packet[12];
+			uint16_t flagId = m_Packet[13] << 8 | m_Packet[14];
+			bool newState = m_Packet[15] > 0;
+			if(sensorId == UINT8_MAX) {
+				for (auto& sensor : sensors) {
+					sensor->setFlag(flagId, newState);
+				}
+			} else {
+				auto & sensors = sensorManager.getSensors();
+				if(sensorId < sensors.size()) {
```

Comment:
should it still send acknowledge on invalid sensor id?

Reply:
what should it do instead? :thinking:

Reply:
At this point, maybe just ignore like too short packet?